File: api/personas/route.py
# api/personas/route.py — GET /api/personas, POST /api/personas
from fastapi import APIRouter
from fastapi.responses import JSONResponse
from .._shared.db import get_cursor
from .._shared.auth import get_current_user
import json
from datetime import datetime

router = APIRouter()

@router.get("/personas")
def list_personas(request):
    try:
        auth = request.headers.get("authorization", "")
        user = get_current_user(auth)
    except ValueError as e:
        return JSONResponse({"error": str(e)}, status_code=401)

    with get_cursor() as c:
        c.execute(
            "SELECT persona_id, name, description, message_count, created_at FROM personas WHERE user_id=? ORDER BY created_at DESC",
            (user["user_id"],)
        )
        rows = c.fetchall()

    return {"personas": [dict(row) for row in rows]}

@router.post("/personas")
def create_persona(request):
    try:
        auth = request.headers.get("authorization", "")
        user = get_current_user(auth)
    except ValueError as e:
        return JSONResponse({"error": str(e)}, status_code=401)

    body = json.loads(request.body.decode())
    name = body.get("name", "").strip()
    if not name:
        return JSONResponse({"error": "name 必填"}, status_code=400)

    import uuid
    persona_id = str(uuid.uuid4())
    description = body.get("description", "")
    extracted_persona = body.get("extracted_persona", {})
    chat_data = body.get("chat_data", {})

    with get_cursor() as c:
        c.execute(
            "INSERT INTO personas (user_id, persona_id, name, description, extracted_persona, chat_data, created_at) VALUES (?, ?, ?, ?, ?, ?, ?)",
            (user["user_id"], persona_id, name, description,
             json.dumps(extracted_persona, ensure_ascii=False),
             json.dumps(chat_data, ensure_ascii=False),
             datetime.utcnow().isoformat())
        )

    return {"persona_id": persona_id, "name": name}, 201

File: frontend/app/api/personas/route.ts
// app/api/personas/route.ts — GET/POST /api/personas
import { NextRequest, NextResponse } from 'next/server'
import { getDb } from '../../_shared/db'
import { getCurrentUser } from '../../_shared/auth'
import crypto from 'crypto'

function uuid() {
  return crypto.randomUUID()
}

export async function GET(request: NextRequest) {
  try {
    const user = getCurrentUser(request.headers.get('authorization'))
    const db = getDb()
    const rows = db.prepare(
      'SELECT persona_id, name, description, message_count, created_at FROM personas WHERE user_id=? ORDER BY created_at DESC'
    ).all(user.user_id) as any[]
    return NextResponse.json({ personas: rows })
  } catch (e: any) {
    const status = e.message?.includes('Token') ? 401 : 500
    return NextResponse.json({ error: e.message }, { status })
  }
}

export async function POST(request: NextRequest) {
  try {
    const user = getCurrentUser(request.headers.get('authorization'))
    const body = await request.json()
    const { name, description, extracted_persona, chat_data } = body as any
    if (!name?.trim()) {
      return NextResponse.json({ error: 'name 必填' }, { status: 400 })
    }
    const personaId = uuid()
    const db = getDb()
    db.prepare(
      `INSERT INTO personas (user_id, persona_id, name, description, extracted_persona, chat_data, created_at)
       VALUES (?, ?, ?, ?, ?, ?, ?)`
    ).run(
      user.user_id,
      personaId,
      name.trim(),
      description || '',
      JSON.stringify(extracted_persona || {}),
      JSON.stringify(chat_data || {}),
      new Date().toISOString()
    )
    return NextResponse.json({ persona_id: personaId, name: name.trim() }, { status: 201 })
  } catch (e: any) {
    return NextResponse.json({ error: e.message }, { status: 500 })
  }
}
